### PR TITLE
Clesaec/tcomp 1950 bom

### PR DESCRIPTION
**Pull request by clesaec**
https://jira.talendforge.org/browse/TCOMP-1950

### Requirements

* Any code change adding any logic **MUST** be tested through a unit test executed with the default build
* Any API addition **MUST** be done with a documentation update if relevant

### CODE REVIEW

**Review by undx — COMMENTED** [review]
My previous comments are still relevant! 
Please, also, don't forget to rebase ;-)

File: Jenkinsfile [hunk 1]
```diff
@@ -174,7 +175,7 @@ spec:
             steps {
                 container('main') {
                     withCredentials([ossrhCredentials, gpgCredentials]) {
-                        sh "mvn deploy $DEPLOY_OPTS $EXTRA_BUILD_ARGS -s .jenkins/settings.xml"
+                        sh "mvn deploy -f ./bom/pom.xml -s .jenkins/settings.xml; mvn deploy $DEPLOY_OPTS $EXTRA_BUILD_ARGS -s .jenkins/settings.xml"
```
[the remark this answers is not in the document]
Comment (clesaec) [reply Jenkinsfile]: changed

File: Jenkinsfile [hunk 2]
```diff
@@ -146,7 +147,7 @@ spec:
             steps {
                 container('main') {
                     withCredentials([ossrhCredentials]) {
-                        sh "mvn clean install $BUILD_ARGS $EXTRA_BUILD_ARGS -s .jenkins/settings.xml"
+                        sh "mvn install -f ./bom/pom.xml && mvn clean install $BUILD_ARGS $EXTRA_BUILD_ARGS -s .jenkins/settings.xml"
```
[the remark this answers is not in the document]
Comment (clesaec) [reply Jenkinsfile]: changed

**Comment by clesaec** [comment]
Changes done

**Review by undx — COMMENTED** [review]
**Why don't you include the bom in reactor?**
- No need to do a `mvn install -f bom/pom.xml` before each local build. This more error-less also.
- Simplify jenkinsfile steps (except java17)
- No changes in `release.sh` script

**From my previous comments**
- I think you forgot ci script like `release.sh`...
- Why haven't you moved parent pom's `dependencyManagement` to bom? We have to manage versions and duplicated artifacts in both places, this is a bit cumbersome.
- Don't forget to rebase ;-)

**Comment by clesaec** [comment]
bom is now declared as a module (not declaring root as parent); so, no need a "-f bom/pom.xml" option.

**Review by undx — APPROVED** [review]
LGTM

**Pull request merged**